### PR TITLE
Don't recompute getMovementLeft() excessively during sort operations.

### PR DESCRIPTION
**Pull request by asvitkine**
## Change Summary & Additional Notes
Don't recompute getMovementLeft() excessively during sort operations.

Use a cache object so that the value for each unit is only computed once during sorting. I noticed this in performance profiles of movement code.

Also add UtilityClass annotation and fix a comment in the PerfTimer class.

### CODE REVIEW

File: lib/java-extras/src/main/java/org/triplea/performance/PerfTimer.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * <p>Example usage with auto-close try block:
  *
  * <pre>{@code
- * try(PerfTimer timer = PerfTimer.startTimer("timer_name_0")) {
+ * try(final PerfTimer timer = PerfTimer.time("timer_name_0")) {
```
Comment (RoiEXLab) [inline lib/java-extras/src/main/java/org/triplea/performance/PerfTimer.java:18]: Is this change intended?

Comment (asvitkine) [reply lib/java-extras/src/main/java/org/triplea/performance/PerfTimer.java:18]: Yes, the previous comment was wrong. Just fixing the comment along the way.

Comment (RoiEXLab) [reply lib/java-extras/src/main/java/org/triplea/performance/PerfTimer.java:18]: The final keyword was left out on purpose. It doesn't do anything at all.
```java
try (PerfTimer timer = PerfTimer.time("timer_name_0")) {
  timer = PerfTimer.time("another timer"); // wont compile, timer is final
}
```

Comment (asvitkine) [reply lib/java-extras/src/main/java/org/triplea/performance/PerfTimer.java]: Sorry, the main purpose of this change is to use `time()` instead of `startTimer()` since the later doesn't exist. Removed the `final` now.

Comment (RoiEXLab) [reply lib/java-extras/src/main/java/org/triplea/performance/PerfTimer.java]: No worries, just a small detail I didn't know myself for a long time :D

**Review by RoiEXLab — APPROVED** [review]
